test/metabase/upload_test.clj

```diff
@@ -1351,35 +1351,29 @@
     (with-uploads-allowed
       (testing "Append should fail if there are extra or missing columns in the CSV file"
         (doseq [[csv-rows error-message]
-                {["_mb_row_id,id,name,extra column one,EXTRA COLUMN TWO"]
-                 (trim-lines "The CSV file contains extra columns that are not in the table:
-                              - extra_column_two
-                              - extra_column_one")
-
-                 [""]
+                {[""]
                  (trim-lines "The CSV file is missing columns that are in the table:
                               - id
                               - name")
 
                  ["_mb_row_id,extra 1, extra 2"]
-                 (trim-lines "The CSV file contains extra columns that are not in the table:
-                              - extra_2
-                              - extra_1
-
-                              The CSV file is missing columns that are in the table:
+                 (trim-lines "The CSV file is missing columns that are in the table:
                               - id
+                              - name")
+
+                 ["_mb_row_id,id, extra 2"]
```

Comment:
Should have a test where one of id or name is provided but not the other, maybe?

Reply:
I'm missing something - this precise case that you're commenting on does provide `id`, but not `name`. How is what you're suggesting different from that?

Reply:
@tsmacdonald I'm going to go ahead and merge this PR. If I legitimately missed something here please let me know and I'll fix it in a follow-up.